pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -169,9 +169,15 @@ func (c *Cluster) Start() error {
 	// If the mgr is newly created, wait for it to start before continuing with the service and
 	// module configuration
 	for _, d := range deploymentsToWaitFor {
-		if err := waitForDeploymentToStart(c.clusterInfo.Context, c.context, d); err != nil {
+		mgrLabel := fmt.Sprintf("%s=%s", controller.DaemonIDLabel, cephv1.KeyMgr)
+		allRunning, err := waitForDeploymentToStart(c.clusterInfo.Context, c.context.Clientset, c.clusterInfo.Namespace, mgrLabel)
+		if err != nil {
 			return errors.Wrapf(err, "failed to wait for mgr %q to start", d.Name)
 		}
+		if !allRunning {
+			logger.Infof("waiting for all mgr pods to be in running state")
```

Comment:
If they're not all running, we need to sleep and try again, then time out after a while.